aws/resource_aws_iam_service_specific_credential.go

```diff
@@ -0,0 +1,184 @@
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 64),
+			},
+			"status": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      iam.StatusTypeActive,
+				ValidateFunc: validation.StringInSlice(iam.StatusType_Values(), false),
+			},
+			"service_password": {
+				Type:      schema.TypeString,
+				Sensitive: true,
+				Computed:  true,
+			},
+			"service_user_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"service_specific_credential_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsIamServiceSpecificCredentialCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).iamconn
+
+	input := &iam.CreateServiceSpecificCredentialInput{
+		ServiceName: aws.String(d.Get("service_name").(string)),
+		UserName:    aws.String(d.Get("user_name").(string)),
+	}
+
+	out, err := conn.CreateServiceSpecificCredential(input)
+	if err != nil {
+		return fmt.Errorf("error creating IAM Service Specific Credential: %w", err)
+	}
+
+	cred := out.ServiceSpecificCredential
+
+	d.SetId(fmt.Sprintf("%s:%s", aws.StringValue(cred.ServiceName), aws.StringValue(cred.UserName)))
+	d.Set("service_password", cred.ServicePassword)
+
+	if v, ok := d.GetOk("status"); ok && v.(string) == iam.StatusTypeInactive {
+		updateInput := &iam.UpdateServiceSpecificCredentialInput{
+			ServiceSpecificCredentialId: cred.ServiceSpecificCredentialId,
+			UserName:                    cred.UserName,
+			Status:                      aws.String(v.(string)),
+		}
+
+		_, err := conn.UpdateServiceSpecificCredential(updateInput)
+		if err != nil {
+			return fmt.Errorf("error settings IAM Service Specific Credential status: %w", err)
+		}
+	}
+
+	return resourceAwsIamServiceSpecificCredentialRead(d, meta)
+}
+
+func resourceAwsIamServiceSpecificCredentialRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).iamconn
+
+	serviceName, userName, err := decodeAwsIamServiceSpecificCredential(d.Id())
+	if err != nil {
+		return err
+	}
+
+	input := &iam.ListServiceSpecificCredentialsInput{
+		ServiceName: aws.String(serviceName),
+		UserName:    aws.String(userName),
+	}
+
+	out, err := conn.ListServiceSpecificCredentials(input)
+	if isAWSErr(err, iam.ErrCodeNoSuchEntityException, "") {
+		log.Printf("[WARN] IAM Service Specific Credential (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("error reading IAM Service Specific Credential (%s): %w", d.Id(), err)
+	}
+
+	if out == nil || len(out.ServiceSpecificCredentials) == 0 {
+		log.Printf("[WARN] IAM Service Specific Credential (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if len(out.ServiceSpecificCredentials) > 1 {
+		return fmt.Errorf("error reading IAM Service Specific Credential: multiple results found, try adjusting search criteria")
+	}
```

Comment:
We should allow for this use case. The ID is already available through `d.Id()` if we use `ServiceSpecificCredentialId`. We'd then cycle through this slice (which length is at most 2) and return that credential.

Reply:
Ill add `ServiceSpecificCredentialId` as part of the id (cant be used alone) but either way this acts more as a guard and this check is implemented in several resources

Reply:
Yeah, I see your point, that the user won't have access to this ID to specify it later on.
Maybe we can pick one of the credentials based whether they are active or not? And if both are active, just pick the latest one (based on `.ServiceSpecificCredentials[].CreatedDate`)?

It seems like the use case for AWS customers to have 2 credentials is to be able to rotate them without disruption, so it would make sense to go with that approach IMHO. What do you think?

Reply:
Other than the "status" there isn't anything else for the customer to in terms of mutation (e.g. permissions or access to other services), since those are determined for the IAM user itself, so the proposed heuristic shouldn't present any major risks AFAICT.